### PR TITLE
change default RF of system_auth to 3

### PR DESCRIPTION
**Pull request by amoskong**
The default RF of system_auth is 1, we might restart nodes during testing.

### CODE REVIEW

File: sdcm/tester.py [hunk 1]
```diff
@@ -216,6 +216,15 @@ def setUp(self):
             self.db_cluster.wait_for_init()
         if self.cs_db_cluster:
             self.cs_db_cluster.wait_for_init()
+
+        self.log.info('change RF of system_auth to 3')
+        # change RF of system_auth to 3
+        node = self.db_cluster.nodes[0]
+        session = self.cql_connection_patient(node)
+        session.execute("ALTER KEYSPACE system_auth WITH replication = {'class': 'org.apache.cassandra.locator.SimpleStrategy', 'replication_factor':3};")
```
Comment (fruch) [inline sdcm/tester.py]: there no way to config it straight from configuration (in scylla.yaml) ?

Comment (amoskong) [reply sdcm/tester.py]: No, system_auth is a system keyspace, the RF can only be changed by an ALERT query.

**Comment by roydahan** [comment]
Amos is this the what we have in the docs?
When we user authorization in SCT, do we prevent other users to login anonymously?
I don't think so.. let's make sure we do it whereever we have it or even by default testing with authorization enabled.

**Comment by roydahan** [comment]
What about adding "authenticator" param to the test.yaml?

**Comment by amoskong** [comment]
If we use authorization in SCT (authenticator: PasswordAuthenticator), other users can't login anonymously.

Do you mean tests/sample.yaml?

**Comment by roydahan** [comment]
Yes, you need to add it to the yaml.

**Comment by amoskong** [comment]
Update:
- introduced system_auth_rf parameter
- set rf of system_auth to 3 by default
- support to disable the rf update by `system_auth_rf: ''`
- rolling_upgrade: enable Authenticator

Please help to review again, thanks.

**Comment by roydahan** [comment]
@amoskong, there is a PR about to be merged by @fruch that sets all configs for tests in a new way.
After the merge, you'll need to refactor this PR to the new config files.
please follow it.

**Comment by amoskong** [comment]
OK I will rebase my PR when his PR is merged.

**Comment by fruch** [comment]
@amoskong #847 was merged

File: tests/sample.yaml [hunk 2]
```diff
@@ -30,6 +30,17 @@ test_sst3: False
 
 hinted_handoff_disabled: false
 
+# Authenticator configuration
+# authenticator: 'PasswordAuthenticator'
+#
+# user/password parameter will be added to cassandra-stress cmdline
```
[the remark this answers is not in the document]
Comment (amoskong) [reply tests/sample.yaml]: I will change the comment to `user/password parameters are used by cql client to connect cql server`

**Comment by amoskong** [comment]
Just updated the PR
- rebased to latest master
- addressed all comments

Please help to review again, thanks.

File: tests/longevity-200GB-48h-verifier-LimitedMonkey-tls.yaml [hunk 3]
```diff
@@ -1,8 +1,8 @@
...
+prepare_write_cmd: "cassandra-stress write cl=ALL n=200200300  -schema 'replication(factor=3) compaction(strategy=LeveledCompactionStrategy)' -port jmx=6868 -mode cql3 native -rate threads=1000 -col 'size=FIXED(1024) n=FIXED(1)' -pop seq=1..200200300 -log interval=15"
+prepare_verify_cmd: "cassandra-stress read cl=ALL n=200200300  -schema 'replication(factor=3) compaction(strategy=LeveledCompactionStrategy)' -port jmx=6868 -mode cql3 native -rate threads=2000 -col 'size=FIXED(1024) n=FIXED(1)' -pop seq=1..200200300 -log interval=15"
+stress_cmd: ["cassandra-stress write cl=QUORUM duration=2860m  -schema 'replication(factor=3) compaction(strategy=LeveledCompactionStrategy)' -port jmx=6868 -mode cql3 native -rate threads=250  -col 'size=FIXED(1024) n=FIXED(1)' -pop seq=400200300..600200300 -log interval=15"]
+stress_read_cmd: ["cassandra-stress read cl=ONE duration=2860m -schema 'replication(factor=3) compaction(strategy=LeveledCompactionStrategy)' -port jmx=6868 -mode cql3 native -rate threads=250  -col 'size=FIXED(1024) n=FIXED(1)' -pop seq=1..200200300 -log interval=5"]
 run_fullscan: 'keyspace1.standard1, 5' # 'ks.cf|random, interval(min)''
```
Comment (roydahan) [inline tests/longevity-200GB-48h-verifier-LimitedMonkey-tls.yaml]: In this file you remove the auth but don't add it below.
Is it on purpose?

Comment (amoskong) [reply tests/longevity-200GB-48h-verifier-LimitedMonkey-tls.yaml]: tls is enabled, so Authentication also needs to be enabled.

Had fixed and updated PR.

Comment (roydahan) [reply tests/longevity-200GB-48h-verifier-LimitedMonkey-tls.yaml]: Thanks

**Comment by juliayakovlev** [comment]
> @bentsi we're ok to merge and backport.
> @scylladb/qa-team, please notice that after backport some of the longevities will start using strict authentication.
> It mean you'll need to provide user & password when connecting with cqlsh.
> Also, please report any suspicious issue you may discover.

Which changed should be performed?

**Comment by amoskong** [comment]
We want to enable Authenticator for longevity which enables Authorization. In some jobs, `user` and `password` parameters are required in using cqlsh, and other use case to connect cql server.

**Comment by roydahan** [comment]
> Which changed should be performed?
No change from your side, just be aware.

**Comment by roydahan** [comment]
@bentsi, can you please take care of the backport to 2019.1?
Please note that not all commits should be backported.

**Comment by amoskong** [comment]
This PR contains 3 parts:
1. update rf of system_auth
2. enable authenticator for rolling upgrade
3. enable authenticator for longevity jobs

**Comment by bentsi** [comment]
which commits should be backported?

**Comment by amoskong** [comment]
It's smooth to backport commits to branch-2019.1, but let's delay `enable authenticator for longevity jobs`, we can prepare another new PR for the support, the commits in this PR are not enough.


@bentsi I did a try:
- https://github.com/amoskong/scylla-cluster-tests/commits/branch-2019.1-backport-pr867
- four commits need to be backported to branch-2019.1

**Comment by amoskong** [comment]
ping @bentsi 

Just rebased https://github.com/amoskong/scylla-cluster-tests/commits/branch-2019.1-backport-pr867 to latest origin/branch-2019.1

**Comment by bentsi** [comment]
backported all the commits to `branch-2019.1`.
@amoskong please give proper commit message instead of :
`20d5fee - address review comments (2 weeks ago) <Amos Kong>`
or use `git commit --amend` to amend changes to current commit.

**Comment by amoskong** [comment]
I updated the PR lcally as you requested, but when I rebase the PR, I found the my original PR had been merged to branch-2019.1

File: sdcm/tester.py [hunk 4]
```diff
@@ -221,6 +221,17 @@ def setUp(self):
             self.db_cluster.wait_for_init()
         if self.cs_db_cluster:
             self.cs_db_cluster.wait_for_init()
+
+        # change RF of system_auth
+        system_auth_rf = self.params.get('system_auth_rf', default=3):
```
Comment (bentsi) [inline sdcm/tester.py]: bug!

Comment (amoskong) [reply sdcm/tester.py]: Do you expect the default value is set in sct_config.py?

Comment (amoskong) [reply sdcm/tester.py]: I saw you had backported PR https://github.com/scylladb/scylla-cluster-tests/pull/906 to branch-2019.1, thanks!

**Comment by bentsi** [comment]
we use labels to notify others that the commits were backported

image

**Pull request merged**